python/rikai/types/vision.py

```diff
@@ -40,12 +40,20 @@ def __init__(self, uri: str):
     def __repr__(self) -> str:
         return f"Image(uri={self.uri})"
 
-    def _repr_html_(self):
-        """Jupyter integration
+    def _repr_mimebundle_(self, include=None, exclude=None):
+        from IPython.display import Image
 
-        TODO: find more appropriate way to return image in jupyter?
-        """
-        return f"<img src={self.uri} />"
+        return Image(self.uri)._repr_mimebundle_(include=include, exclude=exclude)
+
+    def _repr_jpeg_(self):
+        from IPython.display import Image
+
+        return Image(self.uri)._repr_jpeg_()
+
+    def _repr_png_(self):
```

Comment:
How are we going to test this?

Not urgent, but what do you think that we extend these core POD class with juptyer in a separate module?

A little bit concern that we will accidentally drag heavy deps like Jupyter into production code.

Reply:
should be easy enough just to put a few small images in the test directory. and we can also add some webtests to refer to web image references (and s3/gs ones)

Reply:
https://ipython.readthedocs.io/en/stable/config/integrating.html#formatters-for-third-party-types  

how about this.  This is basically MVC while these types are `M`, and the extensions are `V`

Reply:
```
The notebook and the Qt console can display richer representations of objects. To use this, you can define any of a number of _repr_*_() methods. Note that these are surrounded by single, not double underscores.
```

isn't that what we're doing?

I added some more tests for these methods just now.

Jupyter: is it that heavy compared to all of the other dependencies in the production system (pytorch/pandas/spark/numpy/etc)

Reply:
my take is that the jupyter integration code is on a separate module to the actual data objects.
Which makes jupyter / display module a `trait` to extend existing codes.

Reply:
we just need the `ipython` module, not all of jupyter. It looks like the package is 1mb according to conda forge. At this stage in the project life-cycle, I would vote that we keep it together. wdyt?